### ISSUE TITLE
[XF] Adapt NavigationStack not possible

### ISSUE THREAD

**Issue by CavalornIV** [PrismLibrary/Prism#798]
### Package info

- Platform: Cross-Platform (Android + iOS)
- Prism version: 6.2.0
- Xamarin version (if applicable): 2.3.2.127

### Description

I have a MainPage from where I can navigate to every view of my solution.
Some views should only be reachable if you're logged in.
So for example if you navigate from MainPage -> ViewA the LoginPage should be displayed before.
After login you should be redirected to ViewA and the LoginPage should be removed from the navigation stack.

This scenario seems to be not possible with the current Prism Navigation:
I try to navigate to the LoginPage by ModalNavigation but to remove the page from the navigation stack I have to GoBack by ModalNavigation too, as far as I know -> The LoginPage redirects me to the MainPage (just for a second) before I can navigate ViewA

Basically I just don't want to navigate to the LoginPage when I press my BackButton on my ViewA.
With the normal Xamarin Forms navigation this seems to be possible due to methods like InsertPageBefore. Is there an equivalent solution in Prism?

### Navigation example

Expected:
MainPage -> LoginPage -> ViewA
and remove the LoginPage from the navigation stack by modal navigation

Current:
MainPage -> LoginPage -> MainPage -> ViewA
I do not want to show the MainPage between the LoginPage and ViewA

Is there an easier way to implement a login flow with Prism Navigation?

**Comment by weases** [comment]
+1

**Comment by powerdude** [comment]
I'd say think differently about the problem.  Go to MainPage -> ViewA and then present the LoginPage modally if the user is not logged in

**Comment by CavalornIV** [comment]
I don't want the user to see ViewA if he's not logged in

Additonally i think if i go from MainPage -> ViewA and present the LoginPage modally and the user cancels the login he will be redirected to ViewA again

**Comment by joacar** [comment]
Why not just present LoginPage if the user is not signed in?

Then navigate to desired view by an absolute uri (which clears the navigation stack and set `Application.Current.MainPage` with the first view in the path of the uri).

Thus `await navService.NavigateAsync("https://host/MainPage/ViewA");` after successful sign in.

**Comment by powerdude** [comment]
agree with @joacar which was going to be my next suggestion.  Do ``` navService.NavigateAsync("/LoginPage")``` first and from login page do ```navService.NavigateAsync("/MainPage/ViewA")```

**Comment by brianlagunas** [comment]
As you can see, there are many options here.  You could either show the login screen first, and if they login then show the protected navigation options/links.  If they don't login the user can still continue, but the protected navigation options/links won't be visible.  You could also juts use an absolute URI which resets the navigation stack, and the deep link to your intended target.
